sdk/core/azure-core/test/ut/CMakeLists.txt

```diff
@@ -39,5 +39,6 @@ target_link_libraries(${TARGET_NAME} PRIVATE azure-core gtest)
 
 # gtest_add_tests will scan the test from azure-core-test and call add_test
 # for each test to ctest. This enables `ctest -r` to run specific tests directly.
-gtest_add_tests(TARGET ${TARGET_NAME})
+gtest_add_tests(TARGET ${TARGET_NAME}
```

Comment:
Instead of this approach, why not remove the prefex and CtestRegex as well?

If we build, per directory/module, then we wouldn't need to worry about all tests running.

Reply:
The CI tests for azure-core must always build all services and run service-tests as well.
This is because a change in Core could break some service.
The problem is when the unit test from some services can't run on CI legs because it requires credentials. That's the case of Storage. 
Then, Azure Core needs to rely on LiveTests to run the Storage tests and on CI, we want to run only Core